mvpa2/generators/partition.py

```diff
@@ -491,6 +493,54 @@ def _get_partition_specs(self, uniqueattrs):
                     if ind < self.count]
 
 
+class FactorialPartitioner(Partitioner):
+    """
+    Given another partitioner on a dataset with two hierarchically organized
+    attributes, generates the combinations of all possible folds of the
+    superordinate attribute with the balanced combinations of the subordinate
+    attribute. For example, splitting familiar and unfamiliar faces while
+    balancing identities in the training/test set.
+
+    ADD EXAMPLE HERE
```

Comment:
YES -- would be super useful.  Verify that it is "runnable" by running doctests

nosetests -s -v --with-doctest --doctest-tests mvpa2/generators/partition.py
```